qiime2/metadata/tests/test_metadata.py

```diff
@@ -413,41 +412,51 @@ def test_no_source_artifacts(self):
         self.assertEqual(self.md.artifacts, ())
 
     def test_add_zero_artifacts(self):
-        self.md._add_artifacts([])
+        self.md._source_artifact = None
 
         self.assertEqual(self.md.artifacts, ())
 
-    def test_add_artifacts(self):
-        # First two artifacts have the same data but different UUIDs.
-        artifact1 = Artifact.import_data('Mapping', {'a': '1', 'b': '3'})
-        self.md._add_artifacts([artifact1])
+    def test_source_artifacts(self):
+        artifact1 = Artifact.import_data('Mapping', {'a': '1', 'b': '2'})
+        md1 = artifact1.view(Metadata)
+
+        artifact2 = Artifact.import_data('Mapping', {'a': '3', 'b': '4'})
+        md2 = artifact2.view(Metadata)
 
-        artifact2 = Artifact.import_data('Mapping', {'a': '1', 'b': '3'})
-        artifact3 = Artifact.import_data('IntSequence1', [1, 2, 3, 4])
-        self.md._add_artifacts([artifact2, artifact3])
+        artifact3 = Artifact.import_data('Mapping', {'c': '5', 'd': '6'})
+        md3 = artifact3.view(Metadata)
+        self.assertEqual(artifact3, md3._source_artifact)
 
-        self.assertEqual(self.md.artifacts, (artifact1, artifact2, artifact3))
+        merged_md = md1.merge(md2, md3)
+        # The artifacts do not come out in a completely deterministic order in
+        # the tuple, it's alphabetical by column name which varies with the
+        # ids
+        self.assertEqual(len(merged_md.artifacts), 3)
+        self.assertIn(artifact1, merged_md.artifacts)
+        self.assertIn(artifact2, merged_md.artifacts)
+        self.assertIn(artifact3, merged_md.artifacts)
 
     def test_add_non_artifact(self):
-        artifact = Artifact.import_data('Mapping', {'a': '1', 'b': '3'})
-
         with self.assertRaisesRegex(TypeError, "Artifact object.*42"):
-            self.md._add_artifacts([artifact, 42])
+            self.md._add_source_artifact(42)
 
         # Test that the object hasn't been mutated.
         self.assertEqual(self.md.artifacts, ())
 
-    def test_add_duplicate_artifact(self):
-        artifact1 = Artifact.import_data('Mapping', {'a': '1', 'b': '3'})
-        artifact2 = Artifact.import_data('IntSequence1', [1, 2, 3, 4])
-        self.md._add_artifacts([artifact1, artifact2])
-
-        with self.assertRaisesRegex(
-                ValueError, "Duplicate source artifacts.*artifact: Mapping"):
-            self.md._add_artifacts([artifact1])
-
-        # Test that the object hasn't been mutated.
-        self.assertEqual(self.md.artifacts, (artifact1, artifact2))
+#    Because of the changed way of tracking source artifacts, this isn't really
```

Comment:
Do this test and it's corresponding test in `test_metadata_column.py` serve a purpose at all now? Given the new way artifacts are tracked, this doesn't really seem relevant anymore.

Reply:
I agree, this seems to not be relevant, now.